src/connection/mod.rs

```diff
@@ -1,36 +1,41 @@
 extern crate dbus;
 
 use std;
-use std::env;
 use general::*;
 use device::DeviceType;
+use manager::NetworkManager;
 
 /// Get a list of Network Manager connections sorted by path.
 ///
 /// # Examples
 ///
 /// ```no_run
 /// use network_manager::connection;
-/// let connections = connection::list().unwrap();
+/// let manager = network_manager::manager::NetworkManager::new();
+/// let connections = connection::list(&manager).unwrap();
 /// println!("{:?}", connections);
 /// ```
-pub fn list() -> Result<Vec<Connection>, String> {
-    let message = dbus_message!(NM_SERVICE_MANAGER,
-                                NM_SETTINGS_PATH,
-                                NM_SETTINGS_INTERFACE,
-                                "ListConnections");
-    let response = dbus_connect!(message);
-    let paths: dbus::arg::Array<dbus::Path, _> = response.get1().unwrap();
-    let mut connections: Vec<_> = paths.map(|p| get_connection(p).unwrap()).collect();
-    connections.sort_by(|a, b| a.cmp(b));
+pub fn list(manager: &NetworkManager) -> Result<Vec<Connection>, String> {
+    let paths = try!(manager.list_connections());
+
+    let mut connections = Vec::new();
+
+    for path in &paths {
+        connections.push(try!(get_connection(&manager, path)))
+    }
+
+    connections.sort();
 
     Ok(connections)
 }
 
 #[test]
 fn test_list_function() {
-    let connections = list().unwrap();
+    let manager = NetworkManager::new();
+
+    let connections = list(&manager).unwrap();
     assert!(connections.len() > 0);
+
     for (index, val) in connections.iter().enumerate() {
         assert_ne!(Connection { ..Default::default() }, val.clone());
         assert_eq!(index as i32, i32::from(val));
```

Comment:
I think you are correct. And I will think about how we can improve some of those test functions a bit further.